src/test/java/edu/ksu/canvas/attendance/submitter/AssignmentSubmitterUTest.java

```diff
@@ -0,0 +1,284 @@
+    private static final Long SECTION_1_ID = 111111111111L;
+    private static final Long SECTION_2_ID = 222222222222L;
+    private static final Long COURSE_ID = 2121212121L;
+    private static final String STUDENT_1_COMMENT = "student 1 comment";
+    private static final String STUDENT_2_COMMENT = "student 2 comment";
+    private static final String STUDENT_3_COMMENT = "student 3 comment";
+    private static final String STUDENT_4_COMMENT = "student 4 comment";
+    private static final String STUDENT_1_NAME = "STUDENT 1 NAME";
+    private static final String STUDENT_2_NAME = "STUDENT 2 NAME";
+    private static final String STUDENT_3_NAME = "STUDENT 3 NAME";
+    private static final String STUDENT_4_NAME = "STUDENT 4 NAME";
+    private static final Long STUDENT_1_ID = 55555555L;
+    private static final Long STUDENT_2_ID = 22222222L;
+    private static final Long STUDENT_3_ID = 55666555L;
+    private static final Long STUDENT_4_ID = 22999222L;
+    private static final Long CANVAS_STUDENT_ID_1 = 1111L;
+    private static final Long CANVAS_STUDENT_ID_2 = 2222L;
+    private static final Long CANVAS_STUDENT_ID_3 = 3333L;
+    private static final Long CANVAS_STUDENT_ID_4 = 4444L;
+    private static final String OAUTH_STRING = "sdfsdfSDFSDFsdfsdFSDFsdfSDFSDgfsdSFDFSDF";
+    private static final long ASSIGNMENT_ID = 8484848484L;
+    private static final long ASSIGNMENT_ID_2 = 77777778484L;
+    private static final String ASSIGNMENT_NAME = "NAME OF THE ASSIGNMENT";
+    private static final String ASSIGNMENT_NAME_2 = "NAME OF THE SECOND ASSIGNMENT";
+    private static final Double ASSIGNMENT_POINTS = 100.0;
+    private static final long CANVAS_ASSIGNMENT_ID = 514514514L;
+    private static final int PROGRESS_ID = 19781971;
+    private static final String SECTION_1_NAME = "SECTION NUMBER 1";
+    private static final String SECTION_2_NAME = "SECTION NUMBER 2";
+
+    private AssignmentSubmitter assignmentSubmitter;
+
+    @Mock
+    private AttendanceAssignmentService assignmentService;
+
+    @Mock
+    private AttendanceSectionService sectionService;
+    //TODO: DONT USE REPOSITORIES AND SERVICES, USE THEIR SERVICES (CALL IN SERVICES IF NEEDED)
+
+    @Mock
+    private AttendanceCourseService attendanceCourseService;
+
+    @Mock
+    private AttendanceService attendanceService;
+
+    @Mock
+    private CanvasApiWrapperService canvasApiWrapperService;
+
+    @Mock
+    private AssignmentValidator assignmentValidator;
+
+    @Mock
+    private CanvasAssignmentAssistant canvasAssignmentAssistant;
+
+
+
+    private NonRefreshableOauthToken oauthToken;
+    private AttendanceSummaryModel attendanceSummaryModel1;
+    private AttendanceSummaryModel attendanceSummaryModel2;
+    private List<AttendanceSummaryModel> attendanceSummaryModelList;
+    private AttendanceAssignment attendanceAssignment1;
+    private AttendanceAssignment attendanceAssignment2;
+    private Optional<Assignment> assignmentOptional;
```

Comment:
I'm conflicted about this. At first it looks like the assignmentOptional is ian optional because sometimes it is empty. But since it is set in the setup method it is never empty. It might be better to use Optional.of(..) in the `thenReturn(..)` calls but I can see that it adds a few calls of `Optional.of(..)`